src/components/node-list/node-list-groups.js

```diff
@@ -1,122 +1,84 @@
-  });
-
   const [collapsed, setCollapsed] = useState(storedState.groupsCollapsed || {});
 
   // Collapse/expand node group
-  const onToggleCollapsed = typeID => {
+  const onToggleGroupCollapsed = typeID => {
     const groupsCollapsed = Object.assign({}, collapsed, {
       [typeID]: !collapsed[typeID]
     });
     setCollapsed(groupsCollapsed);
     saveState({ groupsCollapsed });
   };
 
-  // Toggle node selection depending on whether it's already selected
-  const handleNodeSelection = nodeID => {
-    if (nodeSelected[nodeID]) {
-      onToggleNodeClicked(null);
-    } else {
-      onToggleNodeClicked(nodeID);
-    }
-  };
-
-  const renderTypeGroup = type => {
-    if (!nodes[type.id]) {
-      return null;
-    }
-    return (
-      <NodeListGroup
-        key={type.id}
-        onToggleCollapsed={onToggleCollapsed}
-        type={type}
-        collapsed={collapsed[type.id]}>
-        <ul className="pipeline-nodelist pipeline-nodelist--nested">
-          {nodes[type.id].map(node => (
-            <li key={node.id}>
-              <NodeListRow
-                active={nodeActive[node.id]}
-                checked={!node.disabled_node}
-                disabled={node.disabled_tag || node.disabled_type}
-                id={node.id}
-                label={node.highlightedLabel}
-                name={node.name}
-                onClick={() => handleNodeSelection(node.id)}
-                onMouseEnter={() => onToggleNodeHovered(node.id)}
-                onMouseLeave={() => onToggleNodeHovered(null)}
-                onChange={e => {
-                  onToggleNodesDisabled([node.id], !e.target.checked);
-                }}
-                selected={nodeSelected[node.id]}
-                type={node.type}
-              />
-            </li>
-          ))}
-        </ul>
-      </NodeListGroup>
-    );
-  };
-
-  return (
-    <Flipper flipKey={collapsed}>
-      <ul className="pipeline-nodelist">{types.map(renderTypeGroup)}</ul>
-    </Flipper>
-  );
+  return sections.map(section => (
+    <nav className="pipeline-nodelist-section kedro" key={section.name}>
+      <h4 className="pipeline-nodelist-section__title">{section.name}</h4>
+      <ul className="pipeline-nodelist">
+        {section.types.map(typeId => {
+          const group = groups[typeId];
+          return (
+            <NodeListGroup
+              key={group.id}
+              id={group.id}
+              name={group.name}
+              kind={group.kind}
+              checked={group.checked}
+              childCount={group.count}
+              allUnset={group.allUnset}
+              visibleIcon={group.visibleIcon}
+              invisibleIcon={group.invisibleIcon}
+              collapsed={Boolean(searchValue) ? false : collapsed[group.id]}
+              onToggleCollapsed={onToggleGroupCollapsed}
+              onToggleChecked={onToggleGroupChecked}>
+              <ul className="pipeline-nodelist pipeline-nodelist--nested">
```

Comment:
This element seems like it could be merged with its parent `pipeline-nodelist__children` div, so that it's just a single `<ul>`:
![image](https://user-images.githubusercontent.com/1155816/95448808-ccf1ba80-095b-11eb-8362-a286c0d77e76.png)

I'd probably get rid of the `pipeline-nodelist pipeline-nodelist--nested` classes and just apply that styling to the `pipeline-nodelist__children` class, which currently has no styling.

Reply:
Yeah I see what you mean, I think this one is also pre-existing too. I do still think that tying separate components to a markup structure is generally not a great idea since it limits re-use, as mentioned in my other comment.

Actually, I think this already applies here: the reason this might have been done originally is that `NodeListRow` is used for both for the list items and the group heading which is placed inside a `<h3>`?

Reply:
As above, I'd use a fragment instead of a div if possible? Can the classes be moved from the div to the ul?

Reply:
Yeah again fragments won't work here as classnames are needed on the container unfortunately, but I was able to apply the `container` prop here and removed the extra container.